src/Kestrel.Core/KestrelConfigurationLoader.cs

```diff
@@ -197,24 +201,41 @@ public KestrelConfigurationLoader HandleEndpoint(ulong handle, Action<ListenOpti
             return this;
         }
 
+        // Called from ApplyEndpointDefaults so it applies to even explicit Listen endpoints.
+        // Does not require a call to Load.
+        internal void ApplyConfigurationDefaults(ListenOptions listenOptions)
+        {
+            var defaults = ConfigurationReader.EndpointDefaults;
+
+            var protocols = ParseProtocols(defaults.Protocols);
```

Comment:
So the parsing occurs on loading each listenoption? Why not parse out the protocol in the ConfigurationReader in ReadEndpointDefaults and just do it once?